src/Renderer/UI/SelectedComponentView.fs

```diff
@@ -510,6 +514,64 @@ let private makeNumberOfBitsField model (comp:Component) text dispatch =
                 dispatch ClosePropertiesNotification
     )
 
+let private makeNumberOfInputsField model (comp:Component) dispatch =
+    let sheetDispatch sMsg = dispatch (Sheet sMsg)
+    
+    let title, oldType, nInp =
+        match comp.Type with
+        | GateN (gType, n) -> "Number of inputs", gType, n
+        | c -> failwithf "makeNumberOfInputsField called with invalid component: %A" c
+    
+    let tabHeight = (Browser.Dom.document.getElementById "TabBody").getBoundingClientRect().height
+
+    div [] [
+        Dropdown.dropdown [Dropdown.IsHoverable] [
+            Dropdown.trigger [] [
+                Button.button [Button.Color IsGrey; Button.IsLight] [
+                    str <| sprintf "Change %A gate type" oldType
+                ]
+            ]
+            Dropdown.menu [Props []] [
+                Dropdown.content 
+                    [Props [
+                            Style [Height $"{int (tabHeight/Constants.dropDownHeightFraction)}px";
+                            OverflowY OverflowOptions.Scroll]]]
+                    [
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) And nInp)]]
+                        [str "And"]
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) Or nInp)]]
+                        [str "Or"]
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) Xor nInp)]]
+                        [str "Xor"]
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) Nand nInp)]]
+                        [str "Nand"]
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) Nor nInp)]]
+                        [str "Nor"]
+                    Dropdown.Item.a
+                        [Dropdown.Item.Props [OnClick (fun _ -> model.Sheet.ChangeGate sheetDispatch (ComponentId comp.Id) Xnor nInp)]]
+                        [str "Xnor"]
```

Comment:
Use abstraction here. Map over a list of gate types paired with names the catalog item